src/config.rs

```diff
@@ -34,13 +34,13 @@ enum KeyMaterial {
     Ed25519 {
         private_key: [u8; 32],
         #[serde(skip)]
-        keypair: Option<libp2p::identity::ed25519::Keypair>,
+        keypair: Option<Box<libp2p::identity::ed25519::Keypair>>,
     },
     RsaPkcs8File {
         #[serde(rename = "rsa_pkcs8_filename")]
         filename: String,
         #[serde(skip)]
-        keypair: Option<libp2p::identity::rsa::Keypair>,
+        keypair: Option<Box<libp2p::identity::rsa::Keypair>>,
```

Comment:
made this and above because there was a complaint that the enum variants differ a lot in size but stable rustc cannot tell the sizes (AFAIK) but boxing either did not solve the issue.